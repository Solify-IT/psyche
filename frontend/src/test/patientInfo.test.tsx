import React from 'react';
import renderer from 'react-test-renderer';
import RecordInfo from 'src/components/recordInfo';
import { BrowserRouter } from 'react-router-dom';
import CornerFab from 'src/components/cornerFab';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Button } from '@material-ui/core';
import recordFixture from 'src/fixtures/recordFixture';

Enzyme.configure({ adapter: new Adapter() });

const record = recordFixture;

describe('Patient info', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecordInfo record={record} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders patients info', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RecordInfo record={record} />
      </BrowserRouter>,
    );
    expect(wrapper.find('PatientGeneralInfo')).toHaveLength(3);
  });

  test('renders patient file sections', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RecordInfo record={record} />
      </BrowserRouter>,
    );
    expect(wrapper.find('FormSection')).toHaveLength(2);
  });

  test('renders fab button', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RecordInfo record={record} />
      </BrowserRouter>,
    );
    expect(wrapper.find(CornerFab)).toHaveLength(1);
  });

  test('renders view and modify button per form', () => {
    const wrapper = mount(
      <BrowserRouter>
        <RecordInfo record={record} />
      </BrowserRouter>,
    );
    expect(wrapper.find(Button)).toHaveLength(3 * 2);
  });
});
