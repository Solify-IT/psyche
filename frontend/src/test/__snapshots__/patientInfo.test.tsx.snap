// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient info snapshot renders 1`] = `
<div>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(20px)",
        "transition": "opacity 400ms, transform 400ms",
      }
    }
  >
    <main
      className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12"
      >
        <h1
          className="MuiTypography-root makeStyles-title-6 MuiTypography-h3"
        >
          Expediente
           
          1
        </h1>
        <div
          className="MuiPaper-root makeStyles-patientSection-2 MuiPaper-outlined MuiPaper-rounded"
        >
          <h3
            className="MuiTypography-root makeStyles-patientSectionTitle-3 MuiTypography-h5"
          >
            <strong>
              Paciente #1
            </strong>
          </h3>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Nombre
                  :
                   
                </strong>
                Carlos
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Apellidos
                  :
                   
                </strong>
                Del Rio
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Edad
                  :
                   
                </strong>
                0
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Genero
                  :
                   
                </strong>
                Hombre
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Lugar de nacimiento
                  :
                   
                </strong>
                test
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Fecha de inicio
                  :
                   
                </strong>
                27-04-2021
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Tipo de paciente
                  :
                   
                </strong>
                Joven
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Telefono
                  :
                   
                </strong>
                2126427
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Direccion
                  :
                   
                </strong>
                Temp
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Codigo Postal
                  :
                   
                </strong>
                832032
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root makeStyles-patientSection-2 MuiPaper-outlined MuiPaper-rounded"
        >
          <h3
            className="MuiTypography-root makeStyles-patientSectionTitle-3 MuiTypography-h5"
          >
            <strong>
              Paciente #2
            </strong>
          </h3>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Nombre
                  :
                   
                </strong>
                Carlos
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Apellidos
                  :
                   
                </strong>
                Del Rio
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Edad
                  :
                   
                </strong>
                0
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Genero
                  :
                   
                </strong>
                Hombre
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Lugar de nacimiento
                  :
                   
                </strong>
                test
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Fecha de inicio
                  :
                   
                </strong>
                27-04-2021
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Tipo de paciente
                  :
                   
                </strong>
                Joven
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Telefono
                  :
                   
                </strong>
                2126427
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Direccion
                  :
                   
                </strong>
                Temp
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientSectionRow-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                <strong>
                  Codigo Postal
                  :
                   
                </strong>
                832032
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root makeStyles-patientSection-2 MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <h2
                className="MuiTypography-root makeStyles-patientSectionTitle-3 MuiTypography-h5"
              >
                <strong>
                  Encuesta socioeconomica
                </strong>
              </h2>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientFileRow-5 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
            >
              <h3
                className="MuiTypography-root MuiTypography-h5"
              >
                Encuesta socioeconomica
              </h3>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
            >
              <h4
                className="MuiTypography-root MuiTypography-h5"
              >
                27-04-2021
              </h4>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Consultar
                </span>
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Modificar
                </span>
              </button>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientFileRow-5 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
            >
              <h3
                className="MuiTypography-root MuiTypography-h5"
              >
                Encuesta socioeconomica
              </h3>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
            >
              <h4
                className="MuiTypography-root MuiTypography-h5"
              >
                27-04-2021
              </h4>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Consultar
                </span>
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Modificar
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root makeStyles-patientSection-2 MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <h2
                className="MuiTypography-root makeStyles-patientSectionTitle-3 MuiTypography-h5"
              >
                <strong>
                  Relatoria
                </strong>
              </h2>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-patientFileRow-5 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
            >
              <h3
                className="MuiTypography-root MuiTypography-h5"
              >
                Relatoria
              </h3>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
            >
              <h4
                className="MuiTypography-root MuiTypography-h5"
              >
                27-04-2021
              </h4>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Consultar
                </span>
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Modificar
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(20px)",
        "transition": "opacity 400ms, transform 400ms",
      }
    }
  >
    <a
      aria-disabled={false}
      aria-label="add"
      className="MuiButtonBase-root MuiFab-root makeStyles-cornerFabExtended-8 MuiFab-extended MuiFab-primary"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <p
          className="MuiTypography-root makeStyles-cornerFabText-10 MuiTypography-body1"
        >
          Agregar formato
        </p>
      </span>
    </a>
  </div>
</div>
`;
